app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt

```diff
@@ -187,14 +232,51 @@ class SearchLocationFragment : Fragment() {
         searchLocationViewModel.eventLocations
             .nonNull()
             .observe(this, Observer { list ->
-                list.forEach {
-                    eventLocationList.add(it.name)
-                }
-                adapter.notifyDataSetChanged()
+                popularLocationAdapter.addAll(list.map { it.name })
+                rootView.popularLocationsLayout.isVisible = list.isNotEmpty()
             })
 
-        rootView.eventLocationLv.setOnItemClickListener { parent, view, position, id ->
-            savePlaceAndRedirectToMain(eventLocationList[position])
+        searchLocationViewModel.loadEventsLocation()
+    }
+}
+
+class LocationsAdapter : RecyclerView.Adapter<LocationViewHolder>() {
```

Comment:
IMO it's better to create a new file for a class.

Reply:
@liveHarshit updated